cli/src/main/kotlin/com/wire/kalium/cli/CLIApplication.kt

```diff
@@ -183,19 +181,7 @@ class LoginCommand : CliktCommand(name = "login") {
         }
 
         val userId = coreLogic.globalScope {
-            val sessions = when (val result = this.session.allSessions()) {
-                is GetAllSessionsResult.Success -> result.sessions
-                is GetAllSessionsResult.Failure.NoSessionFound -> emptyList()
-                is GetAllSessionsResult.Failure.Generic -> throw PrintMessage("Failed retrieve existing sessions: ${result.genericFailure}")
-            }
-            if (sessions.map { it.session.userId }.contains(loginResult.session.userId)) {
-                this.session.updateCurrentSession(loginResult.session.userId)
-            } else {
-                val addAccountResult = addAuthenticatedAccount(loginResult, ssoId, true)
-                if (addAccountResult !is AddAuthenticatedUserUseCase.Result.Success) {
-                    throw PrintMessage("Failed to save session")
-                }
-            }
```

Comment:
This logic is handled inside the `addAuthenticatedAccount` use case?

Reply:
Yes, and if there is already a session we decide to replace it or not inside of addAuthenticatedAccountUseCase.